src/core/server/http/lifecycle/auth.ts

```diff
@@ -82,17 +112,42 @@ export interface AuthResultParams {
   responseHeaders?: AuthHeaders;
 }
 
+/**
+ * Result of auth redirection.
+ * @public
+ */
+export interface AuthRedirectedParams {
+  /**
+   * Headers to attach for auth redirect.
+   * Must include "location" header
+   */
+  headers: { location: string } & ResponseHeaders;
+}
+
 /**
  * @public
  * A tool set defining an outcome of Auth interceptor for incoming request.
  */
 export interface AuthToolkit {
   /** Authentication is successful with given credentials, allow request to pass through */
   authenticated: (data?: AuthResultParams) => AuthResult;
+  /**
+   * User has no credentials.
+   * Allows user to access a resource when authRequired: 'optional'
+   * Rejects a request when authRequired: true
+   * */
+  notHandled: () => AuthResult;
+  /**
+   * Redirect user to IdP when authRequired: true
```

Comment:
__typo:__ (IdP is just one case)
```suggestion
   * Redirects user to another location to complete authentication when authRequired: true
```